### PR TITLE
[Documentation] Toasts reference docs

### LINKED ISSUE #1342: [doc]: Toasts documentation
**Describe the request**

There is no guidance on how to use the Toasts feature.

**Possible solutions**

Create documentation for Toasts, which include:

* Clear and understandable JSDocs in source code
* Reference documentation created
* Usage example/tutorial

**Screenshots**

n/a

**Please let us know whether this is a new topic or a topic change request:**
 - [x] New Topic Request (ie. missing entire topic/section)
 - [ ] Topic Change Request (ie. spelling, organization)

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Create documentation for the toasts hooks and venia component

## Related Issue

Closes #1342 

## Verification Steps

1. Navigate to the `pwa-devdocs` directory
2. Build the HTML: `npm run build`
3. Verify the following docs are autogenerated:
    * `src/_includes/auto-generated/venia-ui/lib/components/ToastContainer/toastContainer.md`
    * `src/_includes/auto-generated/peregrine/lib/Toasts/useToasts.md`
    * `src/_includes/auto-generated/peregrine/lib/Toasts/useToastContext.md`
4. Verify the auto-generated docs have content based on the doc blocks in the source files.

**Note:** There is no need to preview server since this content will not be published until the component is published to NPM.

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/components/ToastContainer/toastContainer.js [hunk 1]
```diff
@@ -29,6 +42,16 @@ const ToastContainer = props => {
...
 
+/**
+ * Props for {@link ToastContainer}
+ *
+ * @typedef props
+ *
+ * @property {Object} classes An object containing the class names for the
+ *   ToastContainer and its Toast components
+ * @property {String} classes.root CSS classes for the root container
+ * @property {String} classes.item A currently unused CSS class property
```
Comment (sirugh) [inline packages/venia-ui/lib/components/ToastContainer/toastContainer.js]: If you can't find a reference to this property, in any of the related toast css files, we should just delete the prop from `propTypes`.

File: pwa-devdocs/_drafts/toasts/useToastContext/index.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+## Examples
+
+### Using the Toast context logic
+
+Import the ToastContextProvider and wrap it around components that use Toast data and functionality.
+
+```jsx
+// MyComponent.js
+
+import {ToastContextProvider} from '@peregrine/lib/Toasts/useToastContext'
```
Comment (sirugh) [inline pwa-devdocs/_drafts/toasts/useToastContext/index.md]: This should be `from '@magento/peregrine'`

File: pwa-devdocs/_drafts/toasts/useToastContext/index.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import {ToastContainer, AddToastComponent} from './MyToastComponents'
+
+...
+
+const MyComponent = ()=>{
```
Comment (sirugh) [inline pwa-devdocs/_drafts/toasts/useToastContext/index.md]: How is this code generated? If you wrote this all by hand I can give you my feedback on each place I think needs to be changed individually. Let me know.

File: packages/venia-ui/lib/components/ToastContainer/toastContainer.js (hunk 1 above)
Comment (jcalcaben) [reply packages/venia-ui/lib/components/ToastContainer/toastContainer.js]: Prop deleted

File: pwa-devdocs/_drafts/toasts/useToastContext/index.md (hunk 3 above)
Comment (jcalcaben) [reply pwa-devdocs/_drafts/toasts/useToastContext/index.md]: Code samples fixed.

File: pwa-devdocs/_drafts/toasts/useToastContext/index.md (hunk 2 above)
Comment (jcalcaben) [reply pwa-devdocs/_drafts/toasts/useToastContext/index.md]: fixed

**Pull request merged**